modules/service/src/main/scala/lucuma/odb/service/UserInvitationService.scala

```diff
@@ -164,47 +166,51 @@ object UserInvitationService:
 
   object Statements:
 
-    val createSuperUserInvitation: Query[(User, CreateUserInvitationInput), UserInvitation] =
+    val createInvitationAsSuperUser: Query[(User, CreateUserInvitationInput), UserInvitation] =
       sql"""
         select insert_invitation(
           $user_id,
           $program_id,
           $email_address,
           $program_user_role,
+          ${tag.opt},
           null
         ) from t_program
         where c_program_id = $program_id
       """
         .query(user_invitation)
         .contramap {
-          case (u, CreateUserInvitationInput(pid, e, r)) => (u.id, pid, e, r, pid)
+          case (u, CreateUserInvitationInput.Coi(pid, e, partner)) => (u.id, pid, e, ProgramUserRole.Coi, Tag(partner.tag).some, pid)
+          case (u, CreateUserInvitationInput.CoiRO(pid, e, partner)) => (u.id, pid, e, ProgramUserRole.CoiRO, Tag(partner.tag).some, pid)
+          case (u, CreateUserInvitationInput.Support(pid, e)) => (u.id, pid, e, ProgramUserRole.Support, none, pid)
         }
 
-    val createPiInvitation: Query[(User, Program.Id, EmailAddress, ProgramUserRole), UserInvitation] =
+    val createInvitationAsPi: Query[(User, Program.Id, EmailAddress, ProgramUserRole, Partner), UserInvitation] =
       sql"""
         select insert_invitation(
           $user_id,
           $program_id,
           $email_address,
           $program_user_role,
+          $tag,
```

Comment:
The partner enum in the database turned into a `t_partner` table and I removed the time accounting categories that were mixed in.  It now matches the `lucuma.core.enum.Partner` (which was moved from `core.model`) and there's a `partner` codec.  It may need to go back to being a dynamic enum at some point, but at the moment we can avoid the `Tag` stuff.